### PR TITLE
[ADHOC] feat(sdk): add BoostValidatorEOA enum, default signer to it if none provided during createBoost

### PR DESCRIPTION
**Pull request by sammccord**
🚨 Please review the [guidelines for contributing](https://github.com/boostxyz/boost-protocol/blob/main/.github/CONTRIBUTING.md) to this repository.

### Description

- `BoostValidatorEOA` enum in `Validator` module
- fixing some biome formatting
- use `incentives.length` for `incentiveQuantity` everywhere to help everyone out because I didn't know and others probably didn't either
- general ci cleanup

💔 Thank you!

### CODE REVIEW

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  By using the `[ADHOC]` title prefix, you are bypassing best practice protections.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 66b842fd77ea8442ef279fc05cb7c1b2a729f855
</p>

File: examples/agora-vote/src/agora-vote.test.ts [hunk 1]
```diff
@@ -203,31 +201,31 @@ describe('Boost with Voting Incentive', () => {
...
     const claimDataPayload = await boost.validator.encodeClaimData({
       signer: trustedSigner,
       incentiveData: encodeAbiParameters(
-        [{ name: '', type: 'uint256' }],
+        [{ name: "", type: "uint256" }],
         [rewardAmount],
       ),
       chainId: optimism.id,
-      incentiveQuantity,
+      incentiveQuantity: boost.incentives.length,
```
Comment (Quazia) [inline examples/agora-vote/src/agora-vote.test.ts:228]: 💜

File: packages/sdk/.env.sample [hunk 2]
```diff
@@ -1,5 +1,12 @@
-DEFAULT_CHAIN_ID=31337
-VITE_TEST_NO_DEPLOY_FIXTURES=false
+# Misc. config for local SDK development
+# DEFAULT_CHAIN_ID=31337
+# VITE_TEST_NO_DEPLOY_FIXTURES=false
+
+# Inject Boost validator EOA's
+VITE_BOOST_TESTNET_SIGNER_EOA=0xd663Fd5d490027f07BAFd696Cb03C8d623D3D92b
+VITE_BOOST_MAINNET_SIGNER_EOA=
```
Comment (Quazia) [inline packages/sdk/.env.sample:7]: TBD 😆

File: packages/sdk/src/BoostCore.ts [hunk 3]
```diff
@@ -441,9 +445,23 @@ export class BoostCore extends Deployable<
...
+      const chain = chains.find(({ id }) => id === chainId);
+      if (!chain)
+        throw new InvalidProtocolChainIdError(
+          chainId,
+          chains.map(({ id }) => id),
+        );
       payload.validator = this.SignerValidator({
-        signers: [payload.owner],
+        signers: [
+          (chain.testnet
```
Comment (Quazia) [inline packages/sdk/src/BoostCore.ts:461]: noooice

**Review by Quazia — APPROVED** [review]
LGTM - hard to parse the actual changes through all this ' " gotta figure out where the difference of opinion on linting is coming from cause this makes my eyes cross

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  By using the `[ADHOC]` title prefix, you are bypassing best practice protections.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8562ddf8f4fc5f10526bb99aae1f9b5799791e1c
</p>

**Comment by sammccord** [comment]
> LGTM - hard to parse the actual changes through all this ' " gotta figure out where the difference of opinion on linting is coming from cause this makes my eyes cross

biome precommit formatting should be handling this, double quotes are correct so the fact that we're seeing any singles means something's awry

File: packages/sdk/src/BoostCore.ts [hunk 4]
```diff
@@ -441,9 +445,24 @@ export class BoostCore extends Deployable<
...
+      const testnet = chain.testnet || chain.id === 31337;
       payload.validator = this.SignerValidator({
-        signers: [payload.owner],
+        signers: [
+          (testnet
+            ? BoostValidatorEOA.TESTNET
+            : BoostValidatorEOA.MAINNET) as unknown as Address,
+        ],
         validatorCaller: coreAddress,
       });
```
Comment (jonathandiep) [inline packages/sdk/src/BoostCore.ts:467]: if we are going to reuse the same signers, is it possible if we can pass in existing SignerValidator contracts that have been deployed? I can see possible gas savings

Or is there some consideration for having to create new SignerValidators for each boost?

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  By using the `[ADHOC]` title prefix, you are bypassing best practice protections.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 798145a64a6acbe73d61038f52f8d6cfa8b71a7d
</p>

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  By using the `[ADHOC]` title prefix, you are bypassing best practice protections.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 798145a64a6acbe73d61038f52f8d6cfa8b71a7d
</p>

File: packages/sdk/src/BoostCore.ts (hunk 4 above)
Comment (sammccord) [reply packages/sdk/src/BoostCore.ts:471]: it could definitely be worth referencing one existing validator, but idk how to do that. @topocount has machinations to do it for openallowlist and now we'll throw signervalidator on the pile

**Comment by sammccord** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  By using the `[ADHOC]` title prefix, you are bypassing best practice protections.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against c4bb05b8e7b37ebfcd7c579589ccc905a4d452a6
</p>
